pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/tasks/BuildWheelsTask.groovy

```diff
@@ -114,17 +124,37 @@ class BuildWheelsTask extends DefaultTask {
      */
     private void buildWheels(Project project, Collection<File> installables, PythonDetails pythonDetails) {
 
+        ProgressLoggerFactory progressLoggerFactory = getServices().get(ProgressLoggerFactory)
+        ProgressLogger progressLogger = progressLoggerFactory.newOperation(BuildWheelsTask)
+        progressLogger.setDescription("Building Wheels")
+        progressLogger.started()
+
         WheelExtension wheelExtension = ExtensionUtils.getPythonComponentExtension(project, WheelExtension)
         def pythonExtension = ExtensionUtils.getPythonExtension(project)
 
+        PythonTag pythonTag = PythonTag.findTag(getProject(), getPythonDetails())
+        PlatformTag platformTag = PlatformTag.makePlatformTag(getProject(), getPythonDetails())
+
+        WheelCache wheelCache = new WheelCache(wheelCacheDir, pythonTag, platformTag)
+
+        def taskTimer = new TaskTimer()
+
+        int counter = 0
         installables.each { File installable ->
 
             def packageInfo = PackageInfo.fromPath(installable.path)
             def shortHand = packageInfo.version ? "${packageInfo.name}-${packageInfo.version}" : packageInfo.name
-            def messageHead = 'Preparing wheel ' + shortHand
+
+            def clock = taskTimer.start(shortHand)
+            progressLogger.progress("Preparing wheel $shortHand (${++counter} of ${installables.size()})")
+
+            def wheel = wheelCache.findWheel(packageInfo.name, packageInfo.version, pythonExtension.details.getPythonVersion())
```

Comment:
Let's put this below the `if packageExcludeFilter ...` below to avoid searching for the wheel if the package is going to be excluded anyway.